src/main/python/sweep_table_model.py

```diff
@@ -37,73 +65,146 @@ def connect(self, data: PreFxData):
         ----------
         data : 
             Will be used as the underlying data store. Will emit notifications when 
-            data has been updated. Will recieve notifications when users update 
+            data has been updated. Will receive notifications when users update
             QC states for individual sweeps.
 
         """
 
         data.end_commit_calculated.connect(self.on_new_data)
         self.qc_state_updated.connect(data.on_manual_qc_state_updated)
 
-
     def on_new_data(
         self, 
         sweep_features: List[Dict], 
         sweep_states: List, 
         manual_qc_states: Dict[int, str], 
-        dataset: EphysDataSet
+        data_set: EphysDataSet
     ):
-        """ Called when the underlying data has been completely replaced
+        """ Called when the underlying data has been completely replaced.
+        Clears any old data and populates the table model with new data.
+        Emits new_data signal after the table is done being populated
 
         Parameters
         ----------
-        sweep_features : 
+        sweep_features : List[dict]
             A list of dictionaries. Each element describes a sweep.
-        sweep_states : 
+        sweep_states : List
             A list of dictionaries. Each element contains ancillary information about
             automatic QC results for that sweep.
-        manual_qc_states : 
-            For each sweep, whether the user has manually passed or failed it (or left it untouched).
-        dataset : 
-            The underlying data. Used to extract sweepwise voltage traces
+        manual_qc_states : Dict[int, str]
+            For each sweep, whether the user has manually passed or failed it
+            (or left it untouched)
+        data_set : EphysDataSet
+            The underlying data. Used to extract sweep-wise voltage traces
 
         """
-
-        self.beginRemoveRows(QModelIndex(), 1, self.rowCount())
-        self._data = []
-        self.endRemoveRows()
-
-        state_lookup = {state["sweep_number"]: state for state in sweep_states}
-        plotter = SweepPlotter(dataset, self.plot_config)
-
-        self.beginInsertRows(QModelIndex(), 1, len(sweep_features))
-        for sweep in sorted(sweep_features, key=lambda swp: swp["sweep_number"]):
-
-            sweep_number = sweep["sweep_number"]
-            state = state_lookup[sweep_number]
-
-            test_pulse_plots, experiment_plots = plotter.advance(sweep_number)
-
+        # dictionary of sweep types used for filtering sweeps in table view
+        self.sweep_types: Dict[str, set] = {
```

Comment:
I believe this specific logic does not belong here. The sweep_table_model should know nothing about which properties of sweeps you may want to use to filter data. This class should already receiving the filtered data. @NileGraddis don't you agree?

Reply:
@sgratiy, `sweep_table_model` was changed so that it holds all the sweeps, which allows `sweep_table_view` to quickly hide/show sweeps when filter options are checked. I do agree that `sweep_types` seems a bit out of place here though. Perhaps it would make more sense for `sweep_table_view` to own the attribute `sweep_types`